cpu/arm7_common/Makefile.dep

```diff
@@ -1,4 +1,12 @@
 # use common ARM7 periph code
 USEMODULE += arm7_common_periph
-# use the nano-specs of Newlib when available
-USEMODULE += newlib_nano
+
+ifeq (1,$(PICOLIBC))
```

Comment:
Why are we using a variable here? Shouldn't it be

```
ifeq (picolibc newlib,$(USEMODULE))
# whatever default you want to choose
endif
```

Reply:
Just copied that from `cpu/cortexm_common/Makefile.dep` :)

Do you mean picolibc should be selected by `USEMODULE += picolibc` instead of `PICOLIBC=1`?

Reply:
Yes, I think that would be better, I understand it is unrelated with this PR but it would be good to do this cleanup.

Reply:
Should we do that as a follow-up?
The RISC-V port also uses this variable.

Reply:
I agree that this would be nicer but I think that this should be done in a follow-up, with all affected CPUs changed at once (cortexm, fe310 and lpc23xx).

Reply:
There you go #15011